### DISCUSSION TITLE
multiple m3u on kodi 19 matrix

### DISCUSSION THREAD

**Discussion by zthed** [iptv-org/community#224] [Q&A] [answered]
hallo 

how can i add multiple m3u urls on kodi 19 matrix ?

**Comment by keystroke3** [comment, marked as answer]
Use [PVR IPTV Simple Client](https://kodi.wiki/view/Add-on:PVR_IPTV_Simple_Client). Go the addon settings and write the link there. You can put all the links in an `.m3u8` file and load that instead.

**Reply by zthed** [reply]
thanks i know how i can use m3u url in kodi but i dont know how to make from several m3u urls one m3u8 url ?

**Reply by keystroke3** [reply]
That will depend on the contents of the various links.  Kodi PVR client only likes clean links with direct link to the actual stream. For every link with more than one channel:

If you are on a UNIX System:
- copy all links to a file, `source.txt`
- run the following command to create or update a file called `myiptv.m3u8` containing the links
 ```bash
[ -f myiptv.m3u8 ] && rm myiptv.m3u8;cat source.txt | xargs curl >> myiptv.m3u8 
``` 

If you are on Windows:  
I am sure there is an easy way to automate this with a shell script, maybe someone will help with that.
- Open the link in a browser and download the m3u files. 
- Create a file `myiptv.m3u8`,
- Open each downloaded file and copy everything to the created file.

When done with all files, make sure the file has `#EXTM3U` at the very top of the file. If a link you have is a direct link to a stream, add it to `myiptv.m3u8` with the same formatting as the rest of the links in the file.    
Back in Kodi, you can change the PVR client settings and point it to the `myiptv.m3u8` file location. Reload Kodi and you should have the channels. To update, you basically have to do this again.

If you are doing this for Android TV or Firestick, the simplest way is to create the myiptv.m3u8 file as shown above, and paste or upload to [github gist](gist.github.com). Click on `raw` and copy and paste the link in Kodi. You can make your life easier by shortening the link on [git.io](git.io). To update, simply edit links in the gist. It should automatically update when you load Kodi.  
Contact me if you want assistance with automation.

**Reply by zthed** [reply]
thanks

**Comment by Sphinxroot** [comment]
hi   you can use playlistloader for matrix  in remote mode 

but it wont play  all stream  from here   because some link  have  #EXTVLCOPT:   in it  .

playlistloader support  in m3u / m3u8  list
|User-Agent=
|Referer=
|X-Forwarded-For=
plugin://plugin.video.
plugin://script.module.

but won`t support  request string  from a addon , mostly direct play command  will work

so you can built yourself a fav list of the channel you want  and play it , by using  the raw m3u link   on github in remote  ,then  if you add or change something in it , it will be auto updated  when you reload the list in playlist loader
